skbio/stats/composition.py

```diff
@@ -896,9 +948,27 @@ def ancom(table, grouping,
         else:
             nu = cutoff[4]
         reject = (W >= nu*n_feat)
-    labs = mat.columns
-    return pd.DataFrame({'W': pd.Series(W, index=labs),
-                         'reject': pd.Series(reject, index=labs)})
+
+    feat_ids = mat.columns
+    ancom_df = pd.DataFrame({'W': pd.Series(W, index=feat_ids),
+                             'reject': pd.Series(reject, index=feat_ids)})
+
+    if len(percentiles) == 0:
+        return ancom_df, pd.DataFrame()
+    else:
```

Comment:
After reviewing the code, I think we should leave this as-is for now rather than duplicate the creation of the ``mat`` variable. We'll factor that aspect correctly where we adapt this API for the new contingency/feature table object. Sound ok?